sb3_contrib/common/utils.py

```diff
@@ -0,0 +1,36 @@
+import torch as th
+
+
+def quantile_huber_loss(current_quantiles: th.Tensor, target_quantiles: th.Tensor) -> th.Tensor:
+    """
+    The quantile-regression loss, as described in the QR-DQN and TQC papers.
+    Partially taken from https://github.com/bayesgroup/tqc_pytorch
+
+    :param current_quantiles: current estimate of quantile value,
+        must be either (batch_size, n_quantiles) or (batch_size, n_critics, n_quantiles)
+    :param target_quantiles: target quantile value,
+       must be either (batch_size, n_target_quantiles) or (batch_size, n_critics, n_target_quantiles)
+    :return: the loss
+    """
+    if current_quantiles.ndim != target_quantiles.ndim:
+        raise ValueError(
+            f"Error: The dimension of curremt_quantile ({current_quantiles.ndim}) needs to match "
+            + f"the dimension of target_quantiles ({target_quantiles.ndim})."
+        )
+    if current_quantiles.ndim not in (2, 3):
+        raise ValueError(f"Error: The dimension of current_quantiles ({current_quantiles.ndim}) needs to be either 2 or 3.")
+
+    n_quantiles = current_quantiles.shape[-1]
+
+    # Cumulative probabilities to calculate quantile values.
+    cum_prob = (th.arange(n_quantiles, device=current_quantiles.device, dtype=th.float) + 0.5) / n_quantiles
+    if current_quantiles.ndim == 3:
+        cum_prob = cum_prob.view(1, 1, -1, 1)  # For TQC
```

Comment:
It would be nice to have the meaning of each dimension in comment (as it was done before)
and add the comment on why the shape for TQC is different that for QR-DQN